lambdas/select-cri/src/main/java/uk/gov/di/ipv/core/selectcri/SelectCriHandler.java

```diff
@@ -125,21 +116,43 @@ public APIGatewayProxyResponseEvent handleRequest(
     }
 
     private APIGatewayProxyResponseEvent getNextWebJourneyCri(
-            List<VisitedCredentialIssuerDetailsDto> visitedCredentialIssuers) {
-        if (userHasNotVisited(visitedCredentialIssuers, passportCriId)) {
-            return getJourneyResponse(passportCriId);
+            List<VisitedCredentialIssuerDetailsDto> visitedCredentialIssuers,
+            List<VcStatusDto> currentVcStatuses,
+            String userId)
+            throws ParseException {
+        Optional<APIGatewayProxyResponseEvent> passportResponse =
+                getCriResponse(visitedCredentialIssuers, currentVcStatuses, passportCriId, userId);
+        if (passportResponse.isPresent()) {
+            return passportResponse.get();
         }
 
         if (userHasNotVisited(visitedCredentialIssuers, addressCriId)) {
             return getJourneyResponse(addressCriId);
+        } else {
+            VisitedCredentialIssuerDetailsDto addressVisitDetails =
+                    visitedCredentialIssuers.stream()
+                            .filter(cri -> cri.getCriId().equals(addressCriId))
+                            .findFirst()
+                            .orElseThrow();
+
+            if (!addressVisitDetails.isReturnedWithVc()) {
+                LOGGER.info(
+                        "User has a previous failed visit to address cri due to: {}. Routing user to web journey instead.",
```

Comment:
I think this log line might be a duplicate of the one from the app journey below.

Reply:
Ah yeah looks like it was the wrong message as well. I've updated the message now for both to say its routing the users to the failed journey route